### ISSUE TITLE
[KEYBOARD]: Text Appointments - Checkbox needs instructions or a Submit button

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#2428]
## Description

We are triggering the text appointments to mobile phone action by checking and unchecking a box. This goes against SC 3.2.2 unless we add clear instructions or a Submit button to drive the action. Screenshot attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor, Emily, Peggy_

## Acceptance Criteria

1. As a user, I want to either press a Submit button to turn on/off mobile phone notifications, or read clear instructions that let me know checking/unchecking the box will toggle this behavior.

## WCAG or Vendor Guidance (optional)

* [On Input: Understanding SC 3.2.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html)

## Screenshots or Trace Logs

![Screen Shot 2019-10-14 at 3 08 26 PM](https://user-images.githubusercontent.com/934879/66779938-2cc29c00-ee95-11e9-8f44-d1eb1d8ed622.png)

**Comment by Samara-Strauss** [comment]
Recommendation: Make this read only and require some to click the edit button to edit the field within the mobile phone modal.

**Comment by Samara-Strauss** [comment]
That, or update the copy with directions that meet accessibility needs.

**Comment by lkoenigsberg** [comment]
@elwoodva , please have a look at Samar's recommendations.  Feel free to reach out to her to chat about and decide on an approach.

**Comment by Samara-Strauss** [comment]
This is my suggestion: make the setting read-only in the read-only version of the profile. Edit will happen in the modal (and this works nice since the checkbox is already in the modal). This is prob what we should have done from the beginning since nothing else is editable in the profile without clicking one of the Edit links.

Note: We'll need copy approved for this approach if we decide to go with it. @lkoenigsberg @elwoodva let me know what you think.

<img width="604" alt="Notification design" src="https://user-images.githubusercontent.com/34068740/67100804-5e39b100-f18e-11e9-8002-bd6c6993624a.png">

**Comment by elwoodva** [comment]
@Samara-Strauss @1Copenut

I like Suggestion 1 when the checkbox is there.  But when it's not there it is not clear that you are not opted in.  The words still make you think you're going to get the notifications sent to you.  

If there was an opposite symbol to use when the flag is false that might be better.  

Or if the words without the checkbox were changed and said something like "We won't send..."  But that doesn't really tell you what to do to change it.

I'm going to run this by my team's human factors engineer for review and opinion which will be better than mine and I'll come back to this after that review.

I do think it makes more sense to only be able to edit this in one place if possible.

However, at this point, I can more quickly and easily implement Suggestion 2 if that wording change meets the 508 accessibility criteria.

I'd rather go with something like suggestion 1 even though it is more work because I'd be eliminating a lot of code that does the transaction since it would no longer be needed.

**Comment by Samara-Strauss** [comment]
> I like Suggestion 1 when the checkbox is there. But when it's not there it is not clear that you are not opted in. The words still make you think you're going to get the notifications sent to you.

If someone had not opted in, we either a) wouldn't show any copy there or b) show something like this:

<img width="512" alt="Screen Shot 2019-10-18 at 10 57 49 AM" src="https://user-images.githubusercontent.com/34068740/67105128-2df61080-f196-11e9-84f2-56b5e5ea1745.png">

> However, at this point, I can more quickly and easily implement Suggestion 2 if that wording change meets the 508 accessibility criteria.

Suggestion 2 works for me. All copy needs to be reviewed by our content team but I think any copy they come up with would be in the spirit of what I've suggested.

**Comment by elwoodva** [comment]
Our team has reviewed this and we think it's best to go with Suggestion 1.

However, for consistency we think the second message about not being opted in should also include an icon.  The icon we think would work would be a red 'X'.  Either that or remove the green checkbox when they are opted in.  Again for consistency.

Here is our suggestion for a slight re-wording of the second message:

"You have not opted in to VA health care appointment text message reminders."

Once this is approved we will make the changes.

**Comment by Samara-Strauss** [comment]
OK, I will work with one of our copywriters to get the verbiage right and report back when I have exact copy.

**Comment by elwoodva** [comment]
Sounds good!

**Comment by Samara-Strauss** [comment]
Copy ticket here #2581

**Comment by lkoenigsberg** [comment]
@elwoodva , I think we can solve this using copy instead of making both a copy and design change.

My proposed solution is to use new copy (from https://github.com/department-of-veterans-affairs/va.gov-team/issues/2581) but keep the check boxes on both the profile and within the modal.

cc/ @Samara-Strauss and @claricechan

**Comment by Samara-Strauss** [comment]
@1Copenut — Does this copy get a 👍 from you for the checkbox?

Please check the box if you want to receive text message (SMS) reminders for your VA health care appointments.

**Comment by elwoodva** [comment]
![Screenshot 2019-10-23 at 9.46.33 AM.png](https://images.zenhubusercontent.com/5d30781abe09f9420aada509/035e6582-60d8-4215-8146-65ae03e7a217)

**Comment by 1Copenut** [comment]
@Samara-Strauss This looks good to me. When this feature goes to the 508 office, I'll have them evaluate the language as well.

**Comment by Samara-Strauss** [comment]
Great, thanks @1Copenut!!

@elwoodva mockup looks good to me!

**Comment by Samara-Strauss** [comment]
Closing. Feel free to reopen if you need to.

**Comment by lkoenigsberg** [comment]
Per @1Copenut comment from Oct 23, does this need to go back through 508 review or is it closed and approved?

**Comment by 1Copenut** [comment]
@lkoenigsberg This issue is closed, and was the last one I found in my initial audit. I'll send this over to the VA 508 office and update the Profile Appointment Text Messages status in the [Product Review List](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-product-review-list.md). I'll report back any findings as new issue tickets.
